### ISSUE TITLE
dual stack ingress: support IPAM pools

### ISSUE THREAD

**Issue by guydc** [gardener/gardener-extension-provider-aws#1048]
**How to categorize this issue?**
/area networking
/kind enhancement
/platform aws

**What would you like to be added**:
Support static IPv6 addresses for dualstack LBs. To achieve this, gardener may support allocation of IPv6 CIDRs to the VPC and public subnet from a user-managed IPAM pool. 

**Why is this needed**:
When a user-managed IPAM pool is used, the LB IPv6 addresses are allocated from a stable range that is decoupled from the cluster's lifecycle (not released in case of VPC deletion). Users may also select specific addresses from within these ranges, using the following annotation: `service.beta.kubernetes.io/aws-load-balancer-ipv6-addresses`. These ranges/addresses can be communicated to consumers for allowlisting purposes.

**Comment by yarivlifchuk** [comment]
Yes, this is correct. BYO VPC requires special effort and maintenance. It also deviates from our homogeneous deployments of AWS, GCP, Azure in 30+ regions.
We would like to keep the same IPv6 CIDR block for the LB and also for the egress from the nodes. This is required for the clients and also to the LOBs BE, which accept only traffic coming from UGW.
Annotating an IPAM pool or CIDR Reservation that will be used by Gardener to create the VPC of a new shoot cluster can work for us.
